### ISSUE TITLE
HealthCheckReportCollector fails GetHealthReportAsync on Azure 

### ISSUE THREAD

**Issue by Ericvf** [Xabaril/AspNetCore.Diagnostics.HealthChecks#1779]
Hi -

We are trying to use this on Azure but our applications are detached from the internet using private endpoints. The HTTP request that is supposed to fetch the UIHealthReport from the endpoint is failing. 

> Connection attempts to local addresses (e.g. localhost, 127.0.0.1) and the machine's own IP will fail, except if another process in the same sandbox has created a listening socket on the destination port. The listening port must be > 1024 and not currently in used. Otherwise, you may get below exception.

https://github.com/projectkudu/kudu/wiki/Azure-Web-App-sandbox#local-address-requests

To solve it, I've tried to get the HealthReport without an HTTP request. I was able to do this by injecting `Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckService`.

The code
```csharp
        private async Task<UIHealthReport> GetHealthReportAsync(HealthCheckConfiguration configuration)
        {
            var healthReport = await _healthCheckService.CheckHealthAsync();
            return UIHealthReport.CreateFrom(healthReport);
        }
```

That works. Unfortunately there are little extensibility points that I can hook into. Most interfaces are also internal so I cannot register other implementations for those. The extension methods that binds those together is also using internal classes. 

What can I do to make this work for me? Why is this library using the HTTP endpoint and not the HealthCheckService?

Thanks!

**Comment by sungam3r** [comment]
> Unfortunately there are little extensibility points that I can hook into. Most interfaces are also internal so I cannot register other implementations for those. The extension methods that binds those together is also using internal classes.

Could you please be more descriptive what interfaces you expect to be public and what extensibility points to add?

**Comment by Ericvf** [comment]
For my specific problem we would benefit from `IHealthCheckReportCollector` being made public.

**Comment by sungam3r** [comment]
@Ericvf fixed but I have no idea when it will be published, see #1593.

**Comment by Ericvf** [comment]
Thats nice, thank you for that!!
